src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -162,16 +165,6 @@ private void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    lock (_memoryMappedFile)
-                    {
-                        if (_accessor != null)
```

Comment:
shoudl we not consider still nulling out hte accessor?  And then in CreateReadableStream throwing if this was a disposed object?